pkgs/development/tools/nsis/default.nix

```diff
@@ -28,30 +29,34 @@ stdenv.mkDerivation rec {
   '';
 
   nativeBuildInputs = [ sconsPackages.scons_3_1_2 ];
-  buildInputs = [ zlib ];
+  buildInputs = [ zlib libiconv ];
+
+  preBuild = lib.optionalString stdenv.isLinux ''
+    sconsFlagsArray+=("PATH=$PATH")
+  '' + ''
+    mkdir -p $out/tmp/include
+    cp ${zlib.dev}/include/* $out/tmp/include
+    cp ${libiconv}/include/* $out/tmp/include
+    sconsFlagsArray+=("APPEND_CPPPATH=$out/tmp/include")
+  '';
 
   sconsFlags = [
     "SKIPSTUBS=all"
     "SKIPPLUGINS=all"
     "SKIPUTILS=all"
     "SKIPMISC=all"
-    "APPEND_CPPPATH=${zlib.dev}/include"
```

Comment:
According to
https://github.com/kichik/nsis/blob/c81d5340487ff451ce6776be4fb7fade798797c5/SConstruct#L505
https://scons.org/doc/1.1.0/HTML/scons-user/x526.html
This should be able to take a whitespace-separated list of paths.

If not, you could use `symlinkJoin`.

Reply:
> This should be able to take a whitespace-separated list of paths.

that's the first thing i tried, but it didn't work. i.e. as this is an env variable, I had to wrap the paths in quotes. (e.g. `"path1 path2"`. This led to it trying to use `"path`

Thanks for the pointer to `symlinkJoin`!

Reply:
> This should be able to take a whitespace-separated list of paths.

that's the first thing i tried, but it didn't work. i.e. as this is an env variable, I had to wrap the paths in quotes. (e.g. `"path1 path2"`. This led to it trying to use `"path1` and failing.

Thanks for the pointer to symlinkJoin!